src/dataflow/components/dataflow-program-graph.tsx

```diff
@@ -235,6 +257,22 @@ export class DataflowProgramGraph extends React.Component<IProps, IState> {
           boxWidth: 6,
           fontSize: 12,
           fontFamily: "'Ubuntu', 'Arial', sans-serif"
+        },
+        // TODO: this approach detects the legend click and then keeps
+        // track of which dataset sequence is shown/hidden in the component state.
+        // to prevents error, we must also stop thge subsequent render in shouldComponentUpdate.
+        // in the future, we should investigate an alternative approach that doesn't require
+        // keeping track of chart.js state in our component state.
+        // tslint:disable-next-line
+        onClick: function(e, legendItem) {
+          const index = legendItem.datasetIndex + indexOffset;
+          const defaultLegendClickHandler = Chart.defaults.global.legend && Chart.defaults.global.legend.onClick;
+          if (defaultLegendClickHandler) {
+            defaultLegendClickHandler.call(this, e, legendItem);
+          }
+          if (legendItem.hidden !== undefined) {
+            setDataSetHidden(index, !legendItem.hidden);
+          }
```

Comment:
Not crazy about turning the linter off, but we need a reference to `this` when we call `defaultLegendClickHandler.call(this, e, legendItem);`.  `this` ends up being a reference to a chart element so we would need to do some work to determine how to access that element and pass it to the default click handler if we were to use an arrow function.

Reply:
I just suggested an alternative above.